### ISSUE TITLE
Onepose data capture app issue

### ISSUE THREAD

**Issue by brian2lee** [zju3dv/OnePose_Plus_Plus#33]
I ran into several issues while using the app. First of all, in annotate mode the screen froze several time when I tried to capture the training data. The app isn't stable enough that I'll have to repeat the process many time to generate a success dataset link. Secondly, until now I haven't successfully captured a test data. I followed the instruction mention in the app, but after pressing stop recording it'll just turn back to "ready" without generating any links.

**Comment by hxy-123** [comment]
Hi, can you provide the device type you used for the capture? We have tested the capture app on the iPhone 12 & 14 pro max with lidar and found it works well both in train and test capture mode.

**Comment by brian2lee** [comment]
I'm working on iPad Pro (12.9inch) (3rd gen), ios 15.5, OnePose Capture app ver. 1.4.0.
  Posted an [issue](https://github.com/zju3dv/OnePose/issues/48) about this when I'm trying to work with onepose, decided to change to onepose++. Still using the same setup for data capture. 
  I've already obtained the training data,fixing the issue or another method to create my own test data would be appreciate.
  Sorry for the late reply and possible stupid mistake I've made since I'm new to this and struggling from all kind of problems here and there.

**Comment by AnukritiSinghh** [comment]
hi @brian2lee were you able to obtain the "annotated" data through the app?
I am on iPhone 14 pro but even my screen is freezing. I will greatly appreciate your help.

**Comment by mkatras11** [comment]
Is there any update regarding the test mode of the app? After stopping the  recording it'll just turn back to "ready" and does not generate any links for the captured data.

**Comment by brian2lee** [comment]
@mkatras11 Not yet, still waiting for the developers' reply, if there's any further progress I'll update in the comment. But I'm new to this field so better not put too much hope on me.

**Comment by hxy-123** [comment]
Hi, sorry for the late reply. We are pretty busy on another project. We find that uploading may be unstable due to using app outside China mainland. We are trying to fix the problem!

**Comment by AnukritiSinghh** [comment]
@hxy-123 Thanks for letting us know. 
Can you share exactly what kind of data is needed for "testing"? It will help us in our own dataset preparation. 
Thanks for the help!

**Comment by mkatras11** [comment]
@AnukritiSinghh To my understanding you only need a sequence of rgb images for testing. You can use the annotation tool and keep only the extracted images. However, you don’t get any ground truth information for the testing part, only the prediction.

**Comment by AnukritiSinghh** [comment]
@mkatras11 Thanks for your reply! while running the code it looks like I need intrinsic.txt and other files as well?
did it work for you with just frames?

**Comment by AnukritiSinghh** [comment]
Thanks for your reply @mkatras11, looks like they need frames.txt file as well. Do you know how to get that for test files?
Appreciate your help!!

**Comment by mkatras11** [comment]
If you make a recording with the annotation tool of OnePose cap app, you get a .zip file that includes a file called frames.txt. Then you can use the script called parse_scanned_data.py to process them into a format usable to run inference with OnePose++.

**Comment by AnukritiSinghh** [comment]
Thank you so much!! 
@mkatras11

**Comment by AnukritiSinghh** [comment]
Hi @mkatras11 , 
Do you by any chance if onepose++ or onepose work for multiple objects pose estimation? I mean will it work if I have several objects in my frame?
Thanks!!

**Comment by mkatras11** [comment]
Hi @AnukritiSinghh,
It is indeed possible to make onepose++ work with multiple objects in one scene. Although I haven't personally implemented it myself, I have observed onepose being used in such scenarios. I am confident that the same can be achieved with onepose++ as well. To accomplish this, you would probably need to train an object detector first to identify the specific objects present in the scene. Once the object detector is trained and can successfully recognize the objects of interest and run the inference part of onepose/onepose++ on each of the corresponding objects identified by the detector. Overall, with the right adjustments to the code, this task is achievable. Just keep in mind that training an object detector and integrating it with onepose++ might require some effort, but it is certainly doable.

**Comment by dongho-Han** [comment]
@mkatras11 @brian2lee 
Hello! Can I ask where can I download the onepose cap app? Like @brian2lee said, Opencap app V1.4.0(https://github.com/zju3dv/OnePose_Plus_Plus/issues/33#issuecomment-1594284728)

**Comment by mkatras11** [comment]
@dongho-Han it is available through App Store. Just search for OnePose Capture and you’ll probably find it!

**Comment by dongho-Han** [comment]
@mkatras11 Thanks a lot! It's too sad that the recent version is also v1.4.0...

**Comment by mkatras11** [comment]
@dongho-Han yeah that’s true! However you can get some descent results if you put some effort and time. I got the best results by combining a motion capture system and a high resolution RGB camera (iPhone 15 pro will work well )!
